extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/export/VectorMRExportCommand.java

```diff
@@ -0,0 +1,208 @@
+import org.apache.hadoop.util.Tool;
+import org.apache.hadoop.util.ToolRunner;
+import org.apache.log4j.Logger;
+
+import com.beust.jcommander.JCommander;
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+
+import mil.nga.giat.geowave.adapter.vector.GeotoolsFeatureDataAdapter;
+import mil.nga.giat.geowave.adapter.vector.avro.AvroSimpleFeatureCollection;
+import mil.nga.giat.geowave.adapter.vector.query.cql.CQLQuery;
+import mil.nga.giat.geowave.core.cli.CLIOperationDriver;
+import mil.nga.giat.geowave.core.index.ByteArrayId;
+import mil.nga.giat.geowave.core.store.CloseableIterator;
+import mil.nga.giat.geowave.core.store.adapter.DataAdapter;
+import mil.nga.giat.geowave.core.store.index.Index;
+import mil.nga.giat.geowave.core.store.index.PrimaryIndex;
+import mil.nga.giat.geowave.core.store.query.QueryOptions;
+import mil.nga.giat.geowave.mapreduce.GeoWaveConfiguratorBase;
+import mil.nga.giat.geowave.mapreduce.JobContextAdapterStore;
+import mil.nga.giat.geowave.mapreduce.input.GeoWaveInputFormat;
+
+public class VectorMRExportCommand extends
+		Configured implements
+		Tool,
+		CLIOperationDriver
+{
+	// TODO annotate appropriately when new commandline tools is merged
+	private static final Logger LOGGER = Logger.getLogger(VectorMRExportCommand.class);
+
+	public static final String BATCH_SIZE_KEY = "BATCH_SIZE";
+	private VectorMRExportOptions mrOptions;
+
+	/**
+	 * Main method to execute the MapReduce analytic.
+	 */
+	public int runJob()
+			throws Exception {
+		Configuration conf = super.getConf();
+		if (conf == null) {
+			conf = new Configuration();
+			setConf(conf);
+		}
+		GeoWaveConfiguratorBase.setRemoteInvocationParams(
+				mrOptions.getHdfsHostPort(),
+				mrOptions.getResourceManagerHostPort(),
+				conf);
+		final QueryOptions options = new QueryOptions();
+		final List<String> adapterIds = mrOptions.getAdapterIds();
+		if ((adapterIds != null) && !adapterIds.isEmpty()) {
+			options.setAdapterIds(Lists.transform(
+					adapterIds,
+					new Function<String, ByteArrayId>() {
+
+						@Override
+						public ByteArrayId apply(
+								final String input ) {
+							return new ByteArrayId(
+									input);
+						}
+					}));
+			try (CloseableIterator<DataAdapter<?>> it = options.getAdapters(mrOptions.getAdapterStore())) {
+				while (it.hasNext()) {
+					final DataAdapter<?> adapter = it.next();
+					JobContextAdapterStore.addDataAdapter(
+							conf,
+							adapter);
+				}
+			}
+		}
+		conf.setInt(
+				BATCH_SIZE_KEY,
+				mrOptions.getBatchSize());
+		if (mrOptions.getIndexId() != null) {
+			final Index index = mrOptions.getIndexStore().getIndex(
+					new ByteArrayId(
+							mrOptions.getIndexId()));
+			if (index == null) {
+				JCommander.getConsole().println(
+						"Unable to find index '" + mrOptions.getIndexId() + "' in store");
+				return -1;
+			}
+			if (index instanceof PrimaryIndex) {
+				options.setIndex((PrimaryIndex) index);
```

Comment:
Looking at the GeoWaveInputFormat, I think we have an issue.
I like this approach.  But, it looks like GeoWaveInputFormat overrides this selection.
I think we need a separate ticket to make sure that QueryOptions is not overridden by GeoWaveInputFormat.setIndex and setAdapters.   Basically, the proposal is to have getAdapters  and getIndex on GeoWaveInputFormat used ONLY if the QueryOptions adapters and index  are not set.  Agreed?